extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/IdGenerator.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import java.util.UUID;
+
+public class IdGenerator {
+  private IdGenerator() {}
+
+  public static final IdGenerator idGenerator = new IdGenerator();
+
+  private static final long LONG_MAX_ID = 0x7fffffffffffffffL;
+
+  public long nextId() {
+    // Make sure this is a positive number.
+    // conflicting ids don't get accepted and is enforced by table constraints.
+    return generateSecureRandomUUID().getLeastSignificantBits() & LONG_MAX_ID;
```

Comment:
Using a part of a random UUID here is inefficient. We could add inefficient code as it does not prevent correct operation, indeed, but why do that when we can use a more efficient approach which is equivalent in code complexity? Why not just pull a `long` from `SecureRandom`?

Reply:
Added this 

```
  public long nextId() {
    // Make sure this is a positive number.
    // conflicting ids don't get accepted and is enforced by table constraints.
    return generateSecureRandomLong() & LONG_MAX_ID;
  }

  private long generateSecureRandomLong() {
    SecureRandom secureRandom = new SecureRandom();
    byte[] randomBytes = new byte[8];
    secureRandom.nextBytes(randomBytes);
    ByteBuffer bb = ByteBuffer.wrap(randomBytes);
    return bb.getLong();
  }
}
```